src/server/streamable-http.test.ts

```diff
@@ -0,0 +1,427 @@
+import { IncomingMessage, ServerResponse } from "node:http";
+import { StreamableHTTPServerTransport } from "./streamable-http.js";
+import { JSONRPCMessage } from "../types.js";
+import { Readable } from "node:stream";
+
+// Mock IncomingMessage
+function createMockRequest(options: {
+  method: string;
+  headers: Record<string, string | string[] | undefined>;
+  body?: string;
+}): IncomingMessage {
+  const readable = new Readable();
+  readable._read = () => {};
+  if (options.body) {
+    readable.push(options.body);
+    readable.push(null);
+  }
+
+  return Object.assign(readable, {
+    method: options.method,
+    headers: options.headers,
+  }) as IncomingMessage;
+}
+
+// Mock ServerResponse
+function createMockResponse(): jest.Mocked<ServerResponse> {
+  const response = {
+    writeHead: jest.fn().mockReturnThis(),
+    write: jest.fn().mockReturnThis(),
+    end: jest.fn().mockReturnThis(),
+    on: jest.fn().mockReturnThis(),
+    emit: jest.fn().mockReturnThis(),
+    getHeader: jest.fn(),
+    setHeader: jest.fn(),
+  } as unknown as jest.Mocked<ServerResponse>;
+  return response;
+}
+
+describe("StreamableHTTPServerTransport", () => {
+  const endpoint = "/mcp";
+  let transport: StreamableHTTPServerTransport;
+  let mockResponse: jest.Mocked<ServerResponse>;
+
+  beforeEach(() => {
+    transport = new StreamableHTTPServerTransport(endpoint);
+    mockResponse = createMockResponse();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("Session Management", () => {
+    it("should generate a valid session ID", () => {
+      expect(transport.sessionId).toBeTruthy();
+      expect(typeof transport.sessionId).toBe("string");
+    });
+
+    it("should include session ID in response headers", async () => {
+      const req = createMockRequest({
+        method: "GET",
+        headers: {
+          accept: "text/event-stream"
+        },
+      });
+
+      await transport.handleRequest(req, mockResponse);
+
+      expect(mockResponse.writeHead).toHaveBeenCalledWith(
+        200,
+        expect.objectContaining({
+          "mcp-session-id": transport.sessionId,
+        })
+      );
+    });
+
+    it("should reject invalid session ID", async () => {
+      const req = createMockRequest({
+        method: "GET",
+        headers: {
+          "mcp-session-id": "invalid-session-id",
+        },
+      });
+
+      await transport.handleRequest(req, mockResponse);
+
+      expect(mockResponse.writeHead).toHaveBeenCalledWith(404);
+      // check if the error response is a valid JSON-RPC error format
+      expect(mockResponse.end).toHaveBeenCalledWith(expect.stringContaining('"jsonrpc":"2.0"'));
+      expect(mockResponse.end).toHaveBeenCalledWith(expect.stringContaining('"error"'));
+    });
```

Comment:
Regardless of whether statelessness is in scope or not, I assume this implementation will follow this advice? If so, I think it would be good to have a test case for it:

- If an Mcp-Session-Id is returned by the server during initialization, clients using the Streamable HTTP transport MUST include it in the Mcp-Session-Id header on all of their subsequent HTTP requests.
  - Servers that require a session ID SHOULD respond to requests without an Mcp-Session-Id header (other than initialization) with HTTP 400 Bad Request.